matrix_sdk/src/sas.rs

```diff
@@ -26,7 +28,14 @@ pub struct Sas {
 impl Sas {
     /// Accept the interactive verification flow.
     pub async fn accept(&self) -> Result<()> {
-        if let Some(req) = self.inner.accept() {
+        self.accept_with_settings(Default::default()).await
+    }
+
+    /// Accept the interactive verification flow with specific settings.
+    ///
+    /// Pass `settings` to customizes the accept-request before sending it.
```

Comment:
Let's add a big warning that the settings must be a subset of the offered one, if they aren't a subset of the offered ones the other side will cancel the verification flow. It would be neat if the settings builder would check this but that's probably not worth it considering the limited use-case.

Please document the settings argument under an `# Arguments` section as a markdown list item like so

```markdown

/// # Arguments
///
/// * `foo` - bar
```

An example would be very much welcome as well.

Reply:
I wasn't aware of this subset restriction, so I changed the behaviour to only purge disallowed methods. I think it is better to design it safely than to add a warning and hope for the best.

Doc is done, with example.